src/main/java/io/github/treesitter/jtreesitter/internal/ChainedLibraryLookup.java

```diff
@@ -0,0 +1,29 @@
+package io.github.treesitter.jtreesitter.internal;
+
+import io.github.treesitter.jtreesitter.NativeLibraryLookup;
+import java.lang.foreign.Arena;
+import java.lang.foreign.Linker;
+import java.lang.foreign.SymbolLookup;
+import java.util.Optional;
+import java.util.ServiceLoader;
+
+final class ChainedLibraryLookup implements NativeLibraryLookup {
+    @Override
+    public SymbolLookup get(Arena arena) {
+        var serviceLoader = ServiceLoader.load(NativeLibraryLookup.class);
+        SymbolLookup lookup = (name) -> Optional.empty();
+        for (var libraryLookup : serviceLoader) {
+            lookup = lookup.or(libraryLookup.get(arena));
+        }
+        return lookup.or(findLibrary(arena)).or(Linker.nativeLinker().defaultLookup());
```

Comment:
I don't quite understand what's happening here.

Reply:
Line 14 creates an initial lookup whose `find` method always returns `Optional.empty()`.

Line 15-17 iterates over zero or more user-registered implementations of `NativeLibraryLookup`, calls `NativeLibraryLookup.get(Arena)` for them, and chains them to the initial empty lookup with `SymbolLookup.or()`. Honoring all user-registered implementations is slightly more general than honoring the first one found.

Line 18 chains some fallback lookups; this is from your `TreeSitter.java` patch.

Reply:
Why is line 14 a lambda that returns `Optional<SymbolLookup>`?
Is the `or` actually `SymbolLookup#or` or is it `Optional#or`?